drivers/can/can.c

```diff
@@ -609,24 +609,24 @@ static ssize_t can_read(FAR struct file *filep, FAR char *buffer,
 
       fifo = &reader->fifo;
 
-      while (fifo->rx_head == fifo->rx_tail)
+      if (filep->f_oflags & O_NONBLOCK)
+        {
+          ret = nxsem_trywait(&fifo->rx_sem);
```

Comment:
should we add can_trysem or remove the null wrapper of can_takesem and can_givesem?

Reply:
+1 for removing the unnecessary wrappers if no one's opposed.